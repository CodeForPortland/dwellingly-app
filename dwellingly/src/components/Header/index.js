import React from 'react';
import { Link } from 'react-router-dom';
import dwellinglylogo from '../../assets/images/dwellingly_logo_white.png';

const Header = props => {
  return (

    <header className='App-header'>
    <div className='App-navbar'>
      <ul>
        <li>
          <Link to='/login'>Login</Link>
        </li>
        <li>
          <Link to='/'>Home</Link>
        </li>
      </ul>
    </div>
    </header>
  );
};

export default Header;
