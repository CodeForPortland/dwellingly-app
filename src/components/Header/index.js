import React from "react";
import { Link, useLocation } from "react-router-dom";
import dwellinglylogo from "../../assets/images/dwellingly_logo_white.png";
import LogOutButton from "./LogOutButton";

const Header = (props) => {
  const loc = useLocation();

  if(loc.pathname === "/terms") {
    return null;
  }
	return (
    <header className="navbar bg-gradient">
      <Link className="navbar-item" id="header-logo" to="/">
        <img src={dwellinglylogo} alt="dwellingly logo" />
      </Link>
      <div className="logout-button">
        <LogOutButton />
      </div>
    </header>
  );
};

export default Header;
