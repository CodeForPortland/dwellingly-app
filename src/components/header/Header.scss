header.appHeader{
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  padding: 1rem;
  width: 100%;
}