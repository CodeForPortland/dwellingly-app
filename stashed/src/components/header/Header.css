/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
/* H E R O - H E A D E R S */
/* ==================================================== */
.hero {
  background-image: -webkit-gradient(linear, left top, left bottom, from(#6d8dff), to(#4cb9ef));
  background-image: linear-gradient(180deg, #6d8dff 0%, #4cb9ef 100%);
  color: #ffffff;
  position: relative;
  text-align: center;
}

.hero .actions {
  font-size: 0.875em;
  position: absolute;
  text-transform: uppercase;
  width: 100%;
}

.hero .actions .action {
  height: 3.57143em;
  line-height: 3.57143em;
  padding: 0 1rem;
  position: absolute;
  right: 0;
}

.hero .actions .action--left {
  left: 0;
  right: unset;
}

.hero .actions .icon {
  font-size: 1.5em;
  vertical-align: middle;
}

.hero__wrapper--basic {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  font-size: 0.875em;
  min-height: 3.57143em;
}

.hero__wrapper--basic h2 {
  font-size: 1.25em;
}

.hero__wrapper--contact {
  padding: 1.5em 1em;
}

.hero__wrapper--contact h1 {
  line-height: 1.5em;
  margin-bottom: 0.5em;
}

.hero__wrapper--contact h1 .icon {
  margin-right: 0.25em;
  -webkit-transform: translateY(-2px);
          transform: translateY(-2px);
  vertical-align: middle;
}

.hero__wrapper--contact a {
  color: #ffffff;
}

.hero__wrapper--contact p {
  line-height: 1em;
  margin: 0.25em 0;
}

.hero__wrapper--contact p.secondary {
  font-weight: 500;
}

.hero__wrapper__actions {
  margin-top: 0.5em;
  position: initial;
}

.hero__wrapper__actions .btn {
  background: rgba(255, 255, 255, 0.15);
  font-size: 0.875em;
  margin: 0.5rem 0.25rem 0;
  min-width: 10.28571em;
}

.hero__wrapper__actions .icon {
  font-size: 1em;
  -webkit-transform: translateY(1px);
          transform: translateY(1px);
}

.hero__wrapper + .hero__wrapper--contact {
  padding-top: 0;
}

.hero--form {
  background-image: -webkit-gradient(linear, left top, left bottom, from(#ffaa55), to(#ffbd3b));
  background-image: linear-gradient(180deg, #ffaa55 0%, #ffbd3b 100%);
}

.hero--tenant {
  background-image: -webkit-gradient(linear, left bottom, left top, from(#ff7c5b), to(#ed4c4c));
  background-image: linear-gradient(to top, #ff7c5b 0%, #ed4c4c 100%);
}

/* end hero/headers */
/*# sourceMappingURL=Header.css.map */