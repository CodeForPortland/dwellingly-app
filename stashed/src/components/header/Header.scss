@import "../../constants/constants.scss";

/* H E R O - H E A D E R S */
/* ==================================================== */
.hero {
  background-image: $color__gradient--blue;
  color: $color--white;
  position: relative;
  text-align: center;

  & .actions {
    font-size: emify(14, 16);
    position: absolute;
    text-transform: uppercase;
    width: 100%;

    .action {
      height: emify(50, 14);
      line-height: emify(50, 14);
      padding: 0 1rem;
      position: absolute;
      right: 0;

      &--left {
        left: 0;
        right: unset;
      }
    }

    .icon {
      font-size: 1.5em;
      vertical-align: middle;
    }
  }

  &__wrapper {
    &--basic {
      @include flex_align(center);
      font-size: emify(14, 16);
      min-height: emify(50, 14);

      h2 {
        font-size: emify(20, 16);
      }
    }

    &--contact {
      padding: 1.5em 1em;

      h1 {
        line-height: emify(36, 24);
        margin-bottom: 0.5em;

        .icon {
          margin-right: 0.25em;
          transform: translateY(-2px);
          vertical-align: middle;
        }
      }

      a {
        color: $color__white;
      }

      p {
        line-height: 1em;
        margin: 0.25em 0;

        &.secondary {
          font-weight: 500;
        }
      }
    }

    &__actions {
      margin-top: 0.5em;
      position: initial;
      .btn {
        background: rgba(255, 255, 255, 0.15);
        font-size: emify(14, 16);
        margin: 0.5rem 0.25rem 0;
        min-width: emify(144, 14);
      }

      .icon {
        font-size: 1em;
        transform: translateY(1px);

        &--commentOutline {
          // transform: translateY(-2px);
        }
      }
    }
  }

  &__wrapper + .hero__wrapper--contact {
    padding-top: 0;
  }

  &--form {
    background-image: $color__gradient--yellow;
  }

  &--tenant {
    background-image: $color__gradient--red;
  }
}
/* end hero/headers */