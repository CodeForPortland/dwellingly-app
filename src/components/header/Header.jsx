import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

class Header extends Component {
  static Label({ label, type, children }) {
    return (
      <div className={`hero__wrapper${type && ` hero__wrapper--${type}`}`}>
        {label && <h2>{label}</h2>}
        {children && children()}
      </div>
    );
  }

  render() {
    const { label, type, variant, variants, children } = this.props;
    const variantClasses =
      variants && variants.length > 0
        ? variants.map(variant => `hero--${variant}`).join(' ')
        : '';
    return (
      <header
        className={`hero ${
          variant ? `hero--${variant}` : ''
        } ${variantClasses}`}>
        {label && type && <Header.Label label={label} type={type} />}
        {children && children(...this.props)}
      </header>
    );
  }
}

Header.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.func,
  variant: PropTypes.string,
  variants: PropTypes.arrayOf(PropTypes.string)
};

Header.defaultProps = {
  children: undefined,
  label: undefined,
  type: undefined,
  variant: undefined,
  variants: undefined
};

export default Header;
