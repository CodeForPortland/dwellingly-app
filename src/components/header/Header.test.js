import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header Component', () => {
  it('correctly renders Header component', () => {
    const HeaderComponent = renderer.create(<Header />).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });
});
