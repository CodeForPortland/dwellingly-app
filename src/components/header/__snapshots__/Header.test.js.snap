// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component correctly renders Header component 1`] = `
<header
  className="hero  "
/>
`;
